app/src/org/commcare/dalvik/odk/provider/InstanceProviderAPI.java

```diff
@@ -30,6 +30,12 @@ private InstanceProviderAPI() {}
     public static final String STATUS_COMPLETE = "complete";
     public static final String STATUS_SUBMITTED = "submitted";
     public static final String STATUS_SUBMISSION_FAILED = "submissionFailed";
+
+    /**
+     * Used to signal an instance insertion shouldn't be attached to the
+     * session's form record, but rather create a new record to register with.
+     */
+    public static final String UNINDEXED_SUBMISSION = "unindexedSubmission";
```

Comment:
Not super proud of having to introduce this flag and use it in the form instance insert in FormInstanceXmlParser, but I think it is the most pragmatic option at this point.